name.abuchen.portfolio/src/name/abuchen/portfolio/util/WebAccess.java

```diff
@@ -5,78 +5,214 @@
     {
-        this.scheme = scheme;
+        this.scheme = Objects.requireNonNull(scheme.trim(), "PP WebAccess: scheme is null"); //$NON-NLS-1$
         return this;
     }
 
     public WebAccess addParameter(String param, String value)
     {
+        param = Objects.requireNonNull(param, "PP WebAccess: Parameter is null"); //$NON-NLS-1$
+        value = Objects.requireNonNull(value, "PP WebAccess: Value Agent is null"); //$NON-NLS-1$
         this.parameters.add(new BasicNameValuePair(param, value));
         return this;
     }
 
     public WebAccess addHeader(String param, String value)
     {
+        param = Objects.requireNonNull(param, "PP WebAccess: Header: Parameter is null"); //$NON-NLS-1$
+        value = Objects.requireNonNull(value, "PP WebAccess: Header: Value Agent is null"); //$NON-NLS-1$
         this.headers.add(new BasicHeader(param, value));
         return this;
     }
 
+    public WebAccess addUserAgent(String userAgent)
+    {
+        this.userAgent = userAgent;
+        return this;
+    }
+
+    public WebAccess ignoreContentType(Boolean ignoreContentType)
+    {
+        this.ignoreContentType = ignoreContentType;
+        return this;
+    }
+
     public String get() throws IOException
     {
-        CloseableHttpClient client = HttpClientBuilder.create() //
-                        .setDefaultRequestConfig(defaultRequestConfig) //
-                        .setDefaultHeaders(this.headers) //
-                        .setUserAgent(OnlineHelper.getUserAgent()).build();
 
-        URIBuilder uriBuilder = new URIBuilder().setScheme(scheme).setHost(host).setPath(path);
-        uriBuilder.setParameters(this.parameters);
+        String errorURL = null;
+        CloseableHttpResponse response = null;
 
         try
         {
-            URL objectURL = uriBuilder.build().toURL();
-            CloseableHttpResponse response = client.execute(new HttpGet(objectURL.toString()));
-            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK)
-                throw new IOException(objectURL.toString() + " --> " + response.getStatusLine().getStatusCode()); //$NON-NLS-1$
+            CloseableHttpClient client = HttpClientBuilder.create() //
+                            .setDefaultRequestConfig(defaultRequestConfig) //
+                            .setDefaultHeaders(this.headers) //
+                            .setUserAgent(this.userAgent).build();
+
+            if (this.url != null && !this.url.isEmpty())
+            {
+                response = client.execute(new HttpGet(url));
+                errorURL = this.url;
+            }
+            else
+            {
+                URIBuilder uriBuilder = new URIBuilder().setScheme(this.scheme).setHost(this.host).setPath(this.path);
+                uriBuilder.setParameters(this.parameters);
+                URL objectURL = uriBuilder.build().toURL();
+                response = client.execute(new HttpGet(objectURL.toString()));
```

Comment:
Ich glaube das hier können wir vereinfachen: der URIBuilder baut eine URI, die wird in eine URL umgewandelt, die URL wieder in einen String und HttpGet verwandelt das wieder in eine URI. Man kann die URI auch direkt an HttpGet(Uri) weiterreichen.